accepted/2021/architecture-targeting.md

```diff
@@ -0,0 +1,498 @@
+
+Portable apps are the least well-defined app type, even though it's the current default. On one hand, portable apps are for advanced scenarios that require a single set of assets to be run in multiple production environments. That's not common.
+
+On the other hand, portable apps enable developers to reliably publish apps built on Windows to run on Linux, for example, without any special knowledge. In this case, the "multiple environments" are dev and production. For example, apps that depends on a local database, like [SQLite](https://www.nuget.org/packages/SQLite/) or [Microsoft.Data.SqlClient](https://www.nuget.org/packages/Microsoft.Data.SqlClient/) will run into this scenario, since local databases almost always contain RID-specific assets.
+
+With portable apps, the developer doesn't need to worry about the fact that these database packages include multiple RID implementations. With RID-specific apps, they would.
+
+However, there is more to the story. For example, the `SQLite` package contains (at the time of writing) seven RID-specific implementations (see the following list). They range from `800k` to `4MB`. That means a portable app that uses SQLite will contain all seven copies of the database engine even though it only ever relies on one, and need to accept the significant size increase that comes with that.
+
+* `linux-x64`
+* `osx-x64`
+* `win10-arm`
+* `win10-x64`
+* `win10-x86`
+* `win7-x64`
+* `win7-x86`
+
+See the appendix for an analysis of portable app size metrics.
+
+Portable apps are easier to deploy to production for PaaS scenarios, like Azure Websites. As soon as developers adopt containers, it makes more sense to adopt our [documented RID-specific container patterns](https://github.com/dotnet/dotnet-docker/tree/main/samples/aspnetapp), which results in optimized apps.
+
+The reason to keep portable apps as the default is that it is easier for non-container web developers that target a different operating system than the one they use locally for development. Portable apps are significantly worse for all other scenarios, particularly client.
+
+In short, portable apps should be an opt-in scenario. The user should make a decision on the extra app size and alternate folder structure that they will experience.
+
+Some people may argue that portable apps benefit new developers since they are simpler. That's potentially true, if they are deploying to a PaaS service that runs another operating system or uses a different CPU type. On the other hand, portable apps have the worst metrics (they can be significantly larger) such that their utility for the new developer experience is questionable. They are not our "best foot forward".
+
+## Proposal for behavior changes
+
+There are multiple proposed changes.
+
+### Enable RID-specific as default build type
+
+The default build type should be changed to the RID-specific framework-dependent model. This approach has the following benefits:
```

Comment:
Is the intention to change every existing template? In other words will unit test project not produce RID specific framework model?

Reply:
This isn't a template change. I don't think unit tests care one way or the other. Unless I'm missing something, I'd expect them to be the least impacted.